Add MAAS infrastructure tab for CAPI tutorial (#568)
canonical/k8s-snap: docs/src/capi/tutorial/getting-started.md

```

> You are now all set to deploy the AWS CAPI infrastructure provider.

Reviewer: ```suggestion
You are now all set to deploy the MAAS infrastructure provider.
```

Reviewer 2: That is the tab of the AWS infrastructure provider. So, this should be correct. 
The MAAS one is below.

````

````{group-tab} MAAS

> Start by setting up environment variables defining the MAAS credentials, if
> these are not already defined:

Reviewer 3: ```suggestion
Start by setting up environment variables to allow access to MAAS:
```

```
export MAAS_API_KEY="<maas-api-key>"
export MAAS_ENDPOINT="http://<maas-endpoint>/MAAS"

> export MAAS_DNS_DOMAIN="<maas-dns-domain>"

Reviewer 4: Is there anything more necessary to do if maas_endpoint is `https`?  Is that supported?

Author: AFAIK upstream provider does not suggest/do anything special for `https` endpoints. The provider uses a simple go client which performs http requests through the standard library so I expect things to work properly. Seems like the client uses `InsecureSkipVerify: true` which can be seen [here](https://github.com/spectrocloud/maas-client-go/blob/072142d8aa50f39a4f11cd59e3a83a53f7f047c4/maasclient/client.go#L244).

```
The MAAS infrastructure provider uses these credentials to deploy machines, create DNS records and perform various other operations for workload clusters.

```{warning}
The management cluster needs to be able to resolve dns records from the MAAS domain.
This usually means the management cluster also needs to be deployed on a MAAS machine.
```

> Continue with setting up environment variables defining the machine image and minimum compute resources of the control plane and worker nodes:

Reviewer 3: ```suggestion
Define further environment variables for the machine image and minimum compute resources of the control plane and worker nodes:
```

```
export CONTROL_PLANE_MACHINE_MINCPU="2"
export CONTROL_PLANE_MACHINE_MINMEMORY="4096"
export CONTROL_PLANE_MACHINE_IMAGE="ubuntu"

export WORKER_MACHINE_MINCPU="2"

> export WORKER_MACHINE_MINMEMORY="4096"

Reviewer: I think 4G of memory is a bit high. I'd set it to 2048 by default and the user can choose if they want more.

export WORKER_MACHINE_IMAGE="ubuntu"
```

> Optional environment variables can be defined for specifying resource pools and machine tags like:

Reviewer 3: ```suggestion
Optional environment variables can be defined for specifying resource pools and machine tags:
```
